Automattic/wp-calypso
# Full Site Editing: Add Template Part CPT and Template Part Type taxonomy

#### Changes proposed in this Pull Request

To improve the distinction between different templating elements, we have decided to introduce a new custom post type with a new custom taxonomy.

- Introduce the Template Part custom post type (`wp_template_part`). It contains all the elements that will be inserted through the Template Part block (`a8c/template`) into a Template post type (`wp_template`).
E.g. the Template Parts "Site Header" and "Site Footer" will be inserted into the Template "Homepage".
With this new distinction, by default Content Slot blocks (`a8c/post-content`) are only enabled for Templates (`wp_template`), and not for Template Parts (`wp_template_part`).

- Update the Template Part block (`a8c/template`) in order to take advantage of the new post type.
Now the selector only searches for Template Parts (`wp_template_part`), so that it's not possible to have a Template (`wp_template`) containing another Template.

- Introduce the Template Part Type taxonomy (`wp_template_part_type`). It will be used to tell different Template Parts (`wp_template_part`) apart.
One of the purposes will be, in the long term, to have a way to determine if and where to apply filters and hooks.
Right now we have "Header" and "Footer" planned, but nothing makes use of this yet.

#### Testing instructions



* Follow the [FSE readme](https://github.com/Automattic/wp-calypso/blob/e759e056360b72576e124989c4174afef78bc1db/apps/full-site-editing/README.md#local-development); setup and enable the plugin.
* Create a few Template Parts, such as a site header and a footer.
* Make sure the Template Part block only searches for Template Parts, and not for Templates.
* Make sure the Content Slot block is not available.
* Create or edit a Template.
* Insert some Template Part blocks, and make sure they only search for Template Parts, and not for Templates.
* Make sure the Content Slot block is available.
